app/src/main/java/com/ashlett/memory/Item.kt

```diff
@@ -1,6 +1,6 @@
 package com.ashlett.memory
 
-data class Item(
+class Item(
```

Comment:
This looks like a model and as such I would stick with a `data class`. They are specifically designed for this use case and will help us later when we tackle unit testing.

Reply:
I changed it to be a normal class for 2 reasons:
1. in `GameLogic.hasPair`, it's easier to check if we're looking at the same item again or its pair. `item2 != item1` is `true` for a normal class (items are different if not the same item) but `false` for a data class (if items have the same data, they are equal). I could work around it by looking at both item and its index in the list, but it would make the code a little bit more complicated.
2. I recall you said in one of the JFs that in a data class, there should in general be only `val` attributes  - a pure data class instance should not change after being created. in this case, `isVisible` is changing throughout the game, so it has to be a `var`. Though maybe that could also be worked around by always inserting a new item in the list instead of modifying existing one (not sure though how that would work with `notifyItemRangeChanged` that I mentioned in another comment)